src/components/ListingDetails/index.tsx

```diff
@@ -0,0 +1,421 @@
+      ))}
+    </InnerFieldContainer>
+  );
+}
+
+export default function ListingDetails({
+  listingData,
+  interpretation = false,
+}: {
+  listingData: Listing;
+  interpretation?: boolean;
+}) {
+  const auth = useAuth();
+  const profile = useProfile();
+
+  const listingFields = useMemo(() => {
+    if (listingData.listing_type === 'CASE') {
+      if (interpretation && listingData.needs_interpreter) {
+        return (
+          <ListingFields
+            fields={caseInterpretationFields}
+            listingData={listingData}
+          />
+        );
+      }
+      return (
+        <ListingFields
+          fields={[
+            ...caseFields,
+            {
+              label: listingData.is_detained
+                ? 'Custody Location'
+                : 'Client Location',
+              getValue: data => data.client_location || 'N/A',
+            },
+          ]}
+          listingData={listingData}
+        />
+      );
+    }
+    if (listingData.listing_type === 'INT') {
+      return (
+        <ListingFields
+          fields={interpretationFields}
+          listingData={listingData}
+        />
+      );
+    }
+    if (listingData.listing_type === 'DOC') {
+      return <ListingFields fields={docFields} listingData={listingData} />;
+    }
+    return <ListingFields fields={lcaFields} listingData={listingData} />;
+  }, [listingData, interpretation]);
+
+  const dateComponent = useMemo(() => {
+    if (listingData.listing_type !== 'INT') {
+      let dateHeader = 'Deadline';
+      let dateData = '';
+      if (listingData.listing_type === 'CASE') {
+        dateHeader = 'Next Court/Filing Date';
+        dateData = formatTimestamp(listingData.upcoming_date);
+      } else {
+        dateData = formatTimestamp(listingData.deadline);
+      }
+      return (
+        <IconTextGroup>
+          <Icon type="calendarBig" />
+          <Subheading $bold>{dateHeader}: </Subheading>
+          <Subheading> {dateData}</Subheading>
+        </IconTextGroup>
+      );
+    }
+    return null;
+  }, [listingData]);
+
+  const langSupportLabel = useMemo(() => {
+    let label = '';
+    if (listingData.listing_type === 'INT') {
+      label = 'One-time Interpretation';
+    } else if (listingData.listing_type === 'DOC') {
+      label = 'Document Translation';
+    } else if (interpretation && listingData.listing_type === 'CASE') {
+      label = 'Case Interpretation';
+    } else {
+      return null;
+    }
+    return (
+      <Subheading $color={COLORS.blueMid} $bold>
+        {label}
+      </Subheading>
+    );
+  }, [listingData, interpretation]);
+
+  const Interest = useMemo(() => {
+    if (auth && auth.userId) {
+      return profile?.profileData ? (
+        <InterestForm caseData={listingData as CaseListing} />
+      ) : (
+        <>
+          <H3>
+            Please finish submitting your profile before submitting interest.
+          </H3>
```

Comment:
nit: not sure if it was ever updated since i couldn't find this particular design in the current iterations, but i believe the old text (david era) was "Please complete your profile before submitting interest"

I think that makes more sense semantically than submitting a profile..

Reply:
makes sense; changed!